src/Agent.Plugins/PipelineArtifact/FileContainerProvider.cs

```diff
@@ -30,9 +30,11 @@ internal class FileContainerProvider : IArtifactProvider
         public FileContainerProvider(VssConnection connection, IAppTraceSource tracer)
         {
             BuildHttpClient buildHttpClient = connection.GetClient<BuildHttpClient>();
-            connection = new VssConnection(buildHttpClient.BaseAddress, connection.Credentials);
-            containerClient = connection.GetClient<FileContainerHttpClient>();
-            this.tracer = tracer;
+            using (var connection2 = new VssConnection(buildHttpClient.BaseAddress, connection.Credentials))
```

Comment:
is this a behavior change? I guess not because `connection` was supposed to be const & here and it wasn't :( :(

Reply:
yup, :(